src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private T[] values;
+    private int usedSpace;
 
+    public ArrayList() {
+        values = (T[])new Object[MAX_ITEMS_NUMBER];
+    }
+
+    public void add(T value) {
+        values[usedSpace++] = value;
+        growIfArrayFull();
     }
 
-    @Override
     public void add(T value, int index) {
+        if (index > usedSpace || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add value to "
+                    + "this index because arrayList size is: "
+                    + usedSpace);
+        }
+        growIfArrayFull();
+        addValueToIndex(index, value);
+    }
 
+    private void growIfArrayFull() {
+        if (usedSpace == values.length - 1) {
+            int oldCapacity = values.length;
+            int newCapacity = oldCapacity + (oldCapacity >> 1);
+            Object[] tempValues = new Object[newCapacity];
+            System.arraycopy(values, 0, tempValues, 0, usedSpace);
+            values = (T[])new Object[newCapacity];
+            values = (T[]) tempValues;
+        }
     }
 
-    @Override
-    public void addAll(List<T> list) {
+    private void grow() {
+        int oldCapacity = values.length;
+        int newCapacity = oldCapacity + (oldCapacity >> 1);
+        Object[] tempValues = new Object[newCapacity];
+        System.arraycopy(values, 0, tempValues, 0, usedSpace);
+        values = (T[])new Object[newCapacity];
+        values = (T[]) tempValues;
+    }
 
+    public void addAll(List<T> list) {
+        while ((values.length - usedSpace) < list.size()) {
+            grow();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
-    @Override
     public T get(int index) {
-        return null;
+        if (index >= usedSpace || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Cant get value "
+                    + "of this index because arrayList size is :"
+                    + usedSpace);
+        }
+        return (T) values[index];
     }
 
-    @Override
     public void set(T value, int index) {
-
+        if (index >= usedSpace || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Cant set value "
+                    + "to this index because arrayList size is :"
+                    + usedSpace);
```

Comment:
U use this code 3 times, you can make a method out of it so that you don't have to rewrite it every time and so that your code looks more compact